src/cets_ack.erl

```diff
@@ -0,0 +1,151 @@
+%% CETS Ack Helper Process
+%%
+%% Contains a list of processes, that are waiting for writes to finish.
+%% Collects acks from nodes in the cluster.
+%% When one of the remote nodes go down, the server stops waiting for acks from it.
+-module(cets_ack).
+-behaviour(gen_server).
+
+%% API functions
+-export([
+    start_link/1,
+    set_servers/2,
+    add/2,
+    ack/3,
+    send_remote_down/2
+]).
+
+%% gen_server callbacks
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-include_lib("kernel/include/logger.hrl").
+
+-type ack_pid() :: pid().
+-type server_pid() :: cets:server_pid().
+-type from() :: gen_server:from().
+-type state() :: #{
+    servers := [server_pid()],
+    %% We store tasks directly in the state map
+    from() => [server_pid(), ...]
+}.
+
+-export_type([ack_pid/0]).
+
+%% API functions
+
+-spec start_link(cets:table_name()) -> {ok, ack_pid()}.
+start_link(Tab) ->
+    Name = list_to_atom(atom_to_list(Tab) ++ "_ack"),
+    gen_server:start_link({local, Name}, ?MODULE, [], []).
+
+%% Sets a list of servers to be used for newly added tasks
+-spec set_servers(ack_pid(), [server_pid()]) -> ok.
+set_servers(AckPid, Servers) ->
+    gen_server:cast(AckPid, {set_servers, Servers}),
+    ok.
+
+%% Adds a new task to wait replies
+-spec add(ack_pid(), from()) -> ok.
+add(AckPid, From) ->
+    AckPid ! {add, From},
+    ok.
+
+%% Called by a remote server after an operation is applied.
+-spec ack(ack_pid(), from(), server_pid()) -> ok.
+ack(AckPid, From, RemotePid) ->
+    %% nosuspend makes message sending unreliable
```

Comment:
yeah, it is for the future if someone tried to add nosuspend there.

noconnect is probably fine, because if there is no connection to the node - the servers would receive `DOWN` message from an unreachable server and ack would not be needed to be sent.

nosuspend is usually used when the code is checking that nosuspend is returned and spawns a new process to send a message. But I think we probably would set new async_dist=true flag for the server process, instead of nosuspend option (though in a separate PR).